<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-06-23" />

<title>cellchat analysis - fibroblasts &amp; T cells</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CNS Fibroblast Immune Surveillance</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">cellchat analysis - fibroblasts &amp; T
cells</h1>
<h4 class="date">2023-06-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CNS Fibroblast Immune
Surveillance/</code> <span class="glyphicon glyphicon-question-sign"
aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240724code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240724)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240724code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240724)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomvskipnessCNSFibroblastImmuneSurveillancetreed415bd4eb7d2750ae96f942127dc108bbfdbcc1etargetblankd415bd4a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/vskipness/CNS-Fibroblast-Immune-Surveillance/tree/d415bd4eb7d2750ae96f942127dc108bbfdbcc1e" target="_blank">d415bd4</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomvskipnessCNSFibroblastImmuneSurveillancetreed415bd4eb7d2750ae96f942127dc108bbfdbcc1etargetblankd415bd4a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/vskipness/CNS-Fibroblast-Immune-Surveillance/tree/d415bd4eb7d2750ae96f942127dc108bbfdbcc1e" target="_blank">d415bd4</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/T_cell_annotation.Rmd
    Untracked:  analysis/cellchat_EndC.Rmd
    Untracked:  analysis/cellchat_analysis.Rmd
    Untracked:  analysis/cluster_annotation.Rmd
    Untracked:  analysis/endothelial_cell_annotation.Rmd
    Untracked:  analysis/fibroblast_annotation.Rmd
    Untracked:  analysis/fibroblast_gene_expression_anaylsis.Rmd
    Untracked:  analysis/merge_and_QC.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>load packages</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(RColorBrewer)
  library(viridis)
  library(ggsci)
  library(scater)
  library(scran)
  library(pheatmap)
  library(biomaRt)
  library(fgsea)
  library(grid)
  library(gridExtra)
  library(msigdbr)
  library(muscat)
  library(ggpubr)
  library(ExploreSCdataSeurat3)
  library(enrichplot)
  #library(Matrix.utils)
  library(destiny)
  library(clusterProfiler)
  library(org.Mm.eg.db)
  library(DOSE)
  library(CellChat)
  library(patchwork)
  options(stringsAsFactors = FALSE)
  })</code></pre>
</div>
<div id="load-files" class="section level1">
<h1>load files</h1>
<pre class="r"><code>basedir &lt;- here()
EndC &lt;- readRDS(paste0(basedir, &quot;/data/endothelial_cells_annotated.rds&quot;))
Tcells &lt;- readRDS(paste0(basedir, &quot;/data/T_cells_annotated.rds&quot;))</code></pre>
</div>
<div id="prep-files" class="section level1">
<h1>Prep files</h1>
<pre class="r"><code>Idents(Tcells) &lt;- Tcells$clusterlabel
Tcells &lt;- subset(Tcells, idents = c(&quot;NK cells&quot;, &quot;ILC2&quot;), invert = T)
Tcells$clusterlabel_cc &lt;- &quot;-&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;γδT&quot;)] &lt;- &quot;γδT&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;Tcm CD4/CD8&quot;)] &lt;- &quot;Tcm CD4/CD8&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;Tem CD4&quot;)] &lt;- &quot;Tem CD4&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;Treg&quot;)] &lt;- &quot;Treg&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;Te CD4&quot;)] &lt;- &quot;Te CD4&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;Te CD8&quot;)] &lt;- &quot;Te CD8&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;ISG Te CD8&quot;)] &lt;- &quot;ISG Te CD8&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;Tem CD8&quot;)] &lt;- &quot;Tem CD8&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;Trm CD8&quot;)] &lt;- &quot;Trm CD8&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;ISG Trm CD8&quot;)] &lt;- &quot;ISG Trm CD8&quot;
Tcells$clusterlabel_cc[which(Idents(Tcells)==&quot;Cx3cr1+ CD8&quot;)] &lt;- &quot;Cx3cr1+ CD8&quot;
unique(Tcells$clusterlabel_cc)</code></pre>
<pre><code> [1] &quot;γδT&quot;         &quot;Tem CD4&quot;     &quot;Tem CD8&quot;     &quot;Te CD8&quot;      &quot;ISG Trm CD8&quot; &quot;Tcm CD4/CD8&quot;
 [7] &quot;Treg&quot;        &quot;Cx3cr1+ CD8&quot; &quot;Trm CD8&quot;     &quot;Te CD4&quot;      &quot;ISG Te CD8&quot; </code></pre>
<pre class="r"><code>Idents(EndC) &lt;- EndC$clusterlabel
EndC$clusterlabel_cc &lt;- &quot;-&quot;
EndC$clusterlabel_cc[which(Idents(EndC)==&quot;Capillary&quot;)] &lt;- &quot;Capillary&quot;
EndC$clusterlabel_cc[which(Idents(EndC)==&quot;Venous&quot;)] &lt;- &quot;Venous&quot; 
EndC$clusterlabel_cc[which(Idents(EndC)==&quot;CP EndC&quot;)] &lt;- &quot;CP EndC&quot;
EndC$clusterlabel_cc[which(Idents(EndC)==&quot;Arterial&quot;)] &lt;- &quot;Arterial&quot; 
EndC$clusterlabel_cc[which(Idents(EndC)==&quot;Capillary/Fbs&quot;)] &lt;- &quot;Capillary/Fbs&quot;
unique(EndC$clusterlabel_cc)</code></pre>
<pre><code>[1] &quot;Capillary&quot;     &quot;Capillary/Fbs&quot; &quot;CP EndC&quot;       &quot;Arterial&quot;      &quot;Venous&quot;       </code></pre>
<pre class="r"><code>## merge
seurat &lt;- merge(EndC, Tcells)</code></pre>
</div>
<div id="all-together" class="section level1">
<h1>all together</h1>
<pre class="r"><code>## all
Idents(seurat) &lt;- seurat$clusterlabel_cc
seurat_all &lt;- subset(seurat, downsample = 500)

table(seurat_all$clusterlabel_cc)</code></pre>
<pre><code>
     Arterial     Capillary Capillary/Fbs       CP EndC   Cx3cr1+ CD8    ISG Te CD8   ISG Trm CD8 
          397           500           302           500           500           500           500 
  Tcm CD4/CD8        Te CD4        Te CD8       Tem CD4       Tem CD8          Treg       Trm CD8 
          500           500           500           500           500           316           500 
       Venous           γδT 
          500           221 </code></pre>
<pre class="r"><code>seurat &lt;- NormalizeData(object = seurat_all)</code></pre>
<div id="create-cellchat-object" class="section level2">
<h2>create cellchat object</h2>
<pre class="r"><code>sce &lt;- as.SingleCellExperiment(seurat_all)
rownames(sce) = gsub(&quot;^.*\\.&quot;, &quot;&quot;, rownames(sce))
cellchat &lt;- createCellChat(object = sce, group.by = &quot;clusterlabel_cc&quot;)</code></pre>
<pre><code>[1] &quot;Create a CellChat object from a SingleCellExperiment object&quot;
The `logcounts` assay is used 
The `colData` assay in the SingleCellExperiment object is used as cell meta information 
Set cell identities for the new CellChat object 
The cell groups used for CellChat analysis are  Arterial Capillary Capillary/Fbs CP EndC Cx3cr1+ CD8 ISG Te CD8 ISG Trm CD8 Tcm CD4/CD8 Te CD4 Te CD8 Tem CD4 Tem CD8 Treg Trm CD8 Venous γδT </code></pre>
<pre class="r"><code>levels(cellchat@idents) &lt;- c(&quot;Venous&quot;, &quot;Arterial&quot;, &quot;Capillary/Fbs&quot;, &quot;Capillary&quot;, &quot;CP EndC&quot;, &quot;Treg&quot;, &quot;Te CD4&quot;, &quot;Te CD8&quot;, &quot;ISG Te CD8&quot;, &quot;Tem CD4&quot;, &quot;Tem CD8&quot;, &quot;Trm CD8&quot;, &quot;ISG Trm CD8&quot;, &quot;Tcm CD4/CD8&quot;, &quot;ydT&quot;, &quot;Cx3cr1+ CD8&quot;)</code></pre>
<pre class="r"><code>CellChatDB &lt;- CellChatDB.mouse # use CellChatDB.mouse if running on mouse data

# use all CellChatDB except for &quot;Non-protein Signaling&quot; for cell-cell communication analysis
CellChatDB.use &lt;- CellChatDB
cellchat@DB &lt;- CellChatDB.use</code></pre>
<pre class="r"><code># subset the expression data of signaling genes for saving computation cost
cellchat &lt;- subsetData(cellchat) # This step is necessary even if using the whole database</code></pre>
<pre><code>Issue identified!! Please check the official Gene Symbol of the following genes:  
 H2-BI H2-Ea-ps </code></pre>
<pre class="r"><code>future::plan(&quot;multisession&quot;, workers = 4) # do parallel
cellchat &lt;- identifyOverExpressedGenes(cellchat)
cellchat &lt;- identifyOverExpressedInteractions(cellchat)

# compute communication probability and infer cellular communication network
cellchat &lt;- computeCommunProb(cellchat)</code></pre>
<pre><code>triMean is used for calculating the average gene expression per cell group. 
[1] &quot;&gt;&gt;&gt; Run CellChat on sc/snRNA-seq data &lt;&lt;&lt; [2024-08-09 16:32:44.897161]&quot;
[1] &quot;&gt;&gt;&gt; CellChat inference is done. Parameter values are stored in `object@options$parameter` &lt;&lt;&lt; [2024-08-09 16:36:52.987529]&quot;</code></pre>
<pre class="r"><code>cellchat &lt;- filterCommunication(cellchat, min.cells = 35)
df.net &lt;- subsetCommunication(cellchat)</code></pre>
<pre class="r"><code>cellchat &lt;- computeCommunProbPathway(cellchat, thresh = 1)
cellchat &lt;- aggregateNet(cellchat)

groupSize &lt;- as.numeric(table(cellchat@idents))
par(mfrow = c(1,2), xpd = T)

netVisual_circle(cellchat@net$count, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = &quot;Number of interactions&quot;)
netVisual_circle(cellchat@net$weight, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = &quot;Interaction weights/strength&quot;)</code></pre>
<p><img src="figure/cellchat_EndC.Rmd/communication%20probability-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mat &lt;- cellchat@net$weight
par(mfrow = c(3,4), xpd=TRUE)
for (i in 1:nrow(mat)) {
  mat2 &lt;- matrix(0, nrow = nrow(mat), ncol = ncol(mat), dimnames = dimnames(mat))
  mat2[i, ] &lt;- mat[i, ]
  netVisual_circle(mat2, vertex.weight = groupSize, weight.scale = T, edge.weight.max = max(mat), title.name = rownames(mat)[i])
}</code></pre>
<p><img src="figure/cellchat_EndC.Rmd/communication%20probability-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/cellchat_EndC.Rmd/communication%20probability-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="visualization-of-signalling-pathways" class="section level2">
<h2>visualization of signalling pathways</h2>
<pre class="r"><code># Heatmap

netVisual_heatmap(cellchat, color.heatmap = &quot;RdPu&quot;, measure = &quot;weight&quot;, sources.use = c(&quot;Venous&quot;, &quot;Arterial&quot;, &quot;Capillary/Fbs&quot;, &quot;Capillary&quot;, &quot;CP EndC&quot;), targets.use = c(&quot;Treg&quot;, &quot;Te CD4&quot;, &quot;Te CD8&quot;, &quot;ISG Te CD8&quot;, &quot;Tem CD4&quot;, &quot;Tem CD8&quot;, &quot;Trm CD8&quot;, &quot;ISG Trm CD8&quot;, &quot;Tcm CD4/CD8&quot;, &quot;ydT&quot;, &quot;Cx3cr1+ CD8&quot;))</code></pre>
<p><img src="figure/cellchat_EndC.Rmd/heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>netVisual_heatmap(cellchat, color.heatmap = &quot;RdPu&quot;, measure = &quot;count&quot;, sources.use = c(&quot;Venous&quot;, &quot;Arterial&quot;, &quot;Capillary/Fbs&quot;, &quot;Capillary&quot;, &quot;CP EndC&quot;), targets.use = c(&quot;Treg&quot;, &quot;Te CD4&quot;, &quot;Te CD8&quot;, &quot;ISG Te CD8&quot;, &quot;Tem CD4&quot;, &quot;Tem CD8&quot;, &quot;Trm CD8&quot;, &quot;ISG Trm CD8&quot;, &quot;Tcm CD4/CD8&quot;, &quot;ydT&quot;, &quot;Cx3cr1+ CD8&quot;))</code></pre>
<p><img src="figure/cellchat_EndC.Rmd/heatmap-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Interactions
pairLR.use &lt;- extractEnrichedLR(cellchat, signaling = c(&quot;CCL&quot;,&quot;CXCL&quot;, &quot;LAMININ&quot;, &quot;NOTCH&quot;))

#levels(cellchat@idents)
#netVisual_bubble(cellchat, sources.use = c(&quot;Pericytes&quot;,&quot;Perivascular fibroblasts&quot;, &quot;Vascular smooth muscle cells&quot;, &quot;Meningeal fibroblasts&quot;, &quot;Tem CD4&quot;,&quot;Tem CD8&quot;, &quot;Tcm CD4/CD8&quot;), pairLR.use = pairLR.use , sort.by.source = T,sort.by.target = T, remove.isolate = TRUE)

netVisual_bubble(cellchat, sources.use = c(&quot;Venous&quot;, &quot;Arterial&quot;, &quot;Capillary/Fbs&quot;, &quot;Capillary&quot;, &quot;CP EndC&quot;), targets.use = c(&quot;Treg&quot;, &quot;Te CD4&quot;, &quot;Te CD8&quot;, &quot;ISG Te CD8&quot;, &quot;Tem CD4&quot;, &quot;Tem CD8&quot;, &quot;Trm CD8&quot;, &quot;ISG Trm CD8&quot;, &quot;Tcm CD4/CD8&quot;, &quot;ydT&quot;, &quot;Cx3cr1+ CD8&quot;), pairLR.use = pairLR.use , sort.by.source = T, sort.by.target = T, remove.isolate = TRUE)</code></pre>
<p><img src="figure/cellchat_EndC.Rmd/interactions-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cellchat &lt;- netAnalysis_computeCentrality(cellchat, slot.name = &quot;netP&quot;)

netAnalysis_signalingRole_scatter(cellchat)</code></pre>
<p><img src="figure/cellchat_EndC.Rmd/dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotGeneExpression(cellchat, signaling = &quot;CXCL&quot;, enriched.only = TRUE, type = &quot;violin&quot;)</code></pre>
<p><img src="figure/cellchat_EndC.Rmd/vln%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotGeneExpression(cellchat, signaling = &quot;CCL&quot;, enriched.only = TRUE, type = &quot;violin&quot;)</code></pre>
<p><img src="figure/cellchat_EndC.Rmd/vln%20plot-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sessioninfo" class="section level1">
<h1>sessionInfo</h1>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-apple-darwin20
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.2.0             CellChat_1.6.1              igraph_2.0.3               
 [4] DOSE_3.30.2                 org.Mm.eg.db_3.19.1         AnnotationDbi_1.66.0       
 [7] clusterProfiler_4.12.1      destiny_3.18.0              enrichplot_1.24.2          
[10] ExploreSCdataSeurat3_0.1.0  ggpubr_0.6.0                muscat_1.18.0              
[13] msigdbr_7.5.1               gridExtra_2.3               fgsea_1.30.0               
[16] biomaRt_2.60.1              pheatmap_1.0.12             scran_1.32.0               
[19] scater_1.32.1               scuttle_1.14.0              SingleCellExperiment_1.26.0
[22] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.1       
[25] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[28] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.3.0          
[31] ggsci_3.2.0                 viridis_0.6.5               viridisLite_0.4.2          
[34] RColorBrewer_1.1-3          here_1.0.1                  magrittr_2.0.3             
[37] Seurat_5.1.0                SeuratObject_5.0.2          sp_2.1-4                   
[40] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[43] dplyr_1.1.4                 purrr_1.0.2                 readr_2.1.5                
[46] tidyr_1.3.1                 tibble_3.2.1                ggplot2_3.5.1              
[49] tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] ica_1.0-3                 plotly_4.10.4             zlibbioc_1.50.0          
  [4] tidyselect_1.2.1          bit_4.0.5                 doParallel_1.0.17        
  [7] clue_0.3-65               lattice_0.22-6            rjson_0.2.21             
 [10] blob_1.2.4                rngtools_1.5.2            S4Arrays_1.4.1           
 [13] pbkrtest_0.5.3            parallel_4.4.1            png_0.1-8                
 [16] registry_0.5-1            cli_3.6.3                 ggplotify_0.1.2          
 [19] goftest_1.2-3             VIM_6.2.2                 variancePartition_1.34.0 
 [22] bluster_1.14.0            ggnetwork_0.5.13          BiocNeighbors_1.22.0     
 [25] uwot_0.2.2                shadowtext_0.1.4          curl_5.2.1               
 [28] mime_0.12                 evaluate_0.24.0           tidytree_0.4.6           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           lmerTest_3.1-3            httpuv_1.6.15            
 [37] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [40] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] smoother_1.3              jquerylib_0.1.4           withr_3.0.0              
 [46] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [49] class_7.3-22              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           colourpicker_1.3.0        htmlwidgets_1.6.4        
 [55] fs_1.6.4                  ggrepel_0.9.5             statnet.common_4.9.0     
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] ranger_0.16.0             DESeq2_1.44.0             DEoptimR_1.1-3           
 [64] reticulate_1.38.0         hexbin_1.28.3             zoo_1.8-12               
 [67] XVector_0.44.0            knitr_1.48                network_1.18.2           
 [70] ggplot.multistats_1.0.0   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [73] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [76] caTools_1.18.2            data.table_1.15.4         ggtree_3.12.0            
 [79] RSpectra_0.16-2           irlba_2.3.5.1             ggrastr_1.0.2            
 [82] fastDummies_1.7.3         gridGraphics_0.5-1        lazyeval_0.2.2           
 [85] yaml_2.3.9                survival_3.7-0            scattermore_1.2          
 [88] crayon_1.5.3              RcppAnnoy_0.0.22          progressr_0.14.0         
 [91] tweenr_2.0.3              later_1.3.2               ggridges_0.5.6           
 [94] codetools_0.2-20          GlobalOptions_0.1.2       aod_1.3.3                
 [97] KEGGREST_1.44.1           Rtsne_0.17                shape_1.4.6.1            
[100] limma_3.60.4              estimability_1.5.1        filelock_1.0.3           
[103] pkgconfig_2.0.3           xml2_1.3.6                TMB_1.9.14               
[106] spatstat.univar_3.0-0     EnvStats_2.8.1            scatterplot3d_0.3-44     
[109] aplot_0.2.3               gridBase_0.4-7            spatstat.sparse_3.1-0    
[112] ape_5.8                   xtable_1.8-4              highr_0.11               
[115] car_3.1-2                 plyr_1.8.9                httr_1.4.7               
[118] rbibutils_2.2.16          tools_4.4.1               globals_0.16.3           
[121] beeswarm_0.4.0            broom_1.0.6               nlme_3.1-165             
[124] HDO.db_0.99.1             dbplyr_2.5.0              lme4_1.1-35.5            
[127] digest_0.6.36             numDeriv_2016.8-1.1       Matrix_1.7-0             
[130] farver_2.1.2              tzdb_0.4.0                remaCor_0.0.18           
[133] reshape2_1.4.4            yulab.utils_0.1.5         glue_1.7.0               
[136] cachem_1.1.0              BiocFileCache_2.12.0      polyclip_1.10-7          
[139] generics_0.1.3            Biostrings_2.72.1         ggalluvial_0.12.5        
[142] mvtnorm_1.2-5             parallelly_1.37.1         statmod_1.5.0            
[145] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[148] minqa_1.2.7               pbapply_1.7-2             httr2_1.0.2              
[151] spam_2.10-0               gson_0.1.0                dqrng_0.4.1              
[154] utf8_1.2.4                graphlayouts_1.1.1        gtools_3.9.5             
[157] ggsignif_0.6.4            RcppEigen_0.3.4.0.0       shiny_1.8.1.1            
[160] GenomeInfoDbData_1.2.12   glmmTMB_1.1.9             memoise_2.0.1            
[163] rmarkdown_2.27            scales_1.3.0              svglite_2.1.3            
[166] future_1.33.2             RANN_2.6.1                Cairo_1.6-2              
[169] spatstat.data_3.1-2       rstudioapi_0.16.0         cluster_2.1.6            
[172] spatstat.utils_3.0-5      hms_1.1.3                 fitdistrplus_1.2-1       
[175] munsell_0.5.1             cowplot_1.1.3             colorspace_2.1-0         
[178] FNN_1.1.4                 rlang_1.1.4               xts_0.14.0               
[181] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  dotCall64_1.1-1          
[184] shinydashboard_0.7.2      ggforce_0.4.2             circlize_0.4.16          
[187] laeken_0.5.3              mgcv_1.9-1                xfun_0.46                
[190] sna_2.7-2                 coda_0.19-4.1             e1071_1.7-14             
[193] TH.data_1.1-2             iterators_1.0.14          emmeans_1.10.3           
[196] abind_1.4-5               GOSemSim_2.30.0           treeio_1.28.0            
[199] bitops_1.0-7              Rdpack_2.6                promises_1.3.0           
[202] scatterpie_0.2.3          RSQLite_2.3.7             qvalue_2.36.0            
[205] sandwich_3.1-0            DelayedArray_0.30.1       proxy_0.4-27             
[208] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[211] boot_1.3-30               beachmat_2.20.0           listenv_0.9.1            
[214] Rcpp_1.0.13               edgeR_4.2.1               workflowr_1.7.1          
[217] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-61              
[220] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[223] spatstat.random_3.3-1     R6_2.5.1                  fastmap_1.2.0            
[226] multcomp_1.4-26           fastmatch_1.1-4           rstatix_0.7.2            
[229] TTR_0.24.4                vipor_0.4.7               ROCR_1.0-11              
[232] vcd_1.4-12                rsvd_1.0.5                nnet_7.3-19              
[235] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[238] deldir_2.0-4              ggthemes_5.1.0            htmltools_0.5.8.1        
[241] bit64_4.0.5               spatstat.explore_3.3-1    lifecycle_1.0.4          
[244] blme_1.0-5                nloptr_2.1.1              sass_0.4.9               
[247] vctrs_0.6.5               spatstat.geom_3.3-2       robustbase_0.99-3        
[250] NMF_0.27                  ggfun_0.1.5               future.apply_1.11.2      
[253] bslib_0.7.0               pillar_1.9.0              magick_2.8.4             
[256] pcaMethods_1.96.0         gplots_3.1.3.1            metapod_1.12.0           
[259] locfit_1.5-9.10           jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Fri Aug  9 16:37:08 2024&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-apple-darwin20
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.2.0             CellChat_1.6.1              igraph_2.0.3               
 [4] DOSE_3.30.2                 org.Mm.eg.db_3.19.1         AnnotationDbi_1.66.0       
 [7] clusterProfiler_4.12.1      destiny_3.18.0              enrichplot_1.24.2          
[10] ExploreSCdataSeurat3_0.1.0  ggpubr_0.6.0                muscat_1.18.0              
[13] msigdbr_7.5.1               gridExtra_2.3               fgsea_1.30.0               
[16] biomaRt_2.60.1              pheatmap_1.0.12             scran_1.32.0               
[19] scater_1.32.1               scuttle_1.14.0              SingleCellExperiment_1.26.0
[22] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.1       
[25] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[28] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.3.0          
[31] ggsci_3.2.0                 viridis_0.6.5               viridisLite_0.4.2          
[34] RColorBrewer_1.1-3          here_1.0.1                  magrittr_2.0.3             
[37] Seurat_5.1.0                SeuratObject_5.0.2          sp_2.1-4                   
[40] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[43] dplyr_1.1.4                 purrr_1.0.2                 readr_2.1.5                
[46] tidyr_1.3.1                 tibble_3.2.1                ggplot2_3.5.1              
[49] tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] ica_1.0-3                 plotly_4.10.4             zlibbioc_1.50.0          
  [4] tidyselect_1.2.1          bit_4.0.5                 doParallel_1.0.17        
  [7] clue_0.3-65               lattice_0.22-6            rjson_0.2.21             
 [10] blob_1.2.4                rngtools_1.5.2            S4Arrays_1.4.1           
 [13] pbkrtest_0.5.3            parallel_4.4.1            png_0.1-8                
 [16] registry_0.5-1            cli_3.6.3                 ggplotify_0.1.2          
 [19] goftest_1.2-3             VIM_6.2.2                 variancePartition_1.34.0 
 [22] bluster_1.14.0            ggnetwork_0.5.13          BiocNeighbors_1.22.0     
 [25] uwot_0.2.2                shadowtext_0.1.4          curl_5.2.1               
 [28] mime_0.12                 evaluate_0.24.0           tidytree_0.4.6           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           lmerTest_3.1-3            httpuv_1.6.15            
 [37] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [40] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] smoother_1.3              jquerylib_0.1.4           withr_3.0.0              
 [46] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [49] class_7.3-22              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           colourpicker_1.3.0        htmlwidgets_1.6.4        
 [55] fs_1.6.4                  ggrepel_0.9.5             statnet.common_4.9.0     
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] ranger_0.16.0             DESeq2_1.44.0             DEoptimR_1.1-3           
 [64] reticulate_1.38.0         hexbin_1.28.3             zoo_1.8-12               
 [67] XVector_0.44.0            knitr_1.48                network_1.18.2           
 [70] ggplot.multistats_1.0.0   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [73] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [76] caTools_1.18.2            data.table_1.15.4         ggtree_3.12.0            
 [79] RSpectra_0.16-2           irlba_2.3.5.1             ggrastr_1.0.2            
 [82] fastDummies_1.7.3         gridGraphics_0.5-1        lazyeval_0.2.2           
 [85] yaml_2.3.9                survival_3.7-0            scattermore_1.2          
 [88] crayon_1.5.3              RcppAnnoy_0.0.22          progressr_0.14.0         
 [91] tweenr_2.0.3              later_1.3.2               ggridges_0.5.6           
 [94] codetools_0.2-20          GlobalOptions_0.1.2       aod_1.3.3                
 [97] KEGGREST_1.44.1           Rtsne_0.17                shape_1.4.6.1            
[100] limma_3.60.4              estimability_1.5.1        filelock_1.0.3           
[103] pkgconfig_2.0.3           xml2_1.3.6                TMB_1.9.14               
[106] spatstat.univar_3.0-0     EnvStats_2.8.1            scatterplot3d_0.3-44     
[109] aplot_0.2.3               gridBase_0.4-7            spatstat.sparse_3.1-0    
[112] ape_5.8                   xtable_1.8-4              highr_0.11               
[115] car_3.1-2                 plyr_1.8.9                httr_1.4.7               
[118] rbibutils_2.2.16          tools_4.4.1               globals_0.16.3           
[121] beeswarm_0.4.0            broom_1.0.6               nlme_3.1-165             
[124] HDO.db_0.99.1             dbplyr_2.5.0              lme4_1.1-35.5            
[127] digest_0.6.36             numDeriv_2016.8-1.1       Matrix_1.7-0             
[130] farver_2.1.2              tzdb_0.4.0                remaCor_0.0.18           
[133] reshape2_1.4.4            yulab.utils_0.1.5         glue_1.7.0               
[136] cachem_1.1.0              BiocFileCache_2.12.0      polyclip_1.10-7          
[139] generics_0.1.3            Biostrings_2.72.1         ggalluvial_0.12.5        
[142] mvtnorm_1.2-5             parallelly_1.37.1         statmod_1.5.0            
[145] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[148] minqa_1.2.7               pbapply_1.7-2             httr2_1.0.2              
[151] spam_2.10-0               gson_0.1.0                dqrng_0.4.1              
[154] utf8_1.2.4                graphlayouts_1.1.1        gtools_3.9.5             
[157] ggsignif_0.6.4            RcppEigen_0.3.4.0.0       shiny_1.8.1.1            
[160] GenomeInfoDbData_1.2.12   glmmTMB_1.1.9             memoise_2.0.1            
[163] rmarkdown_2.27            scales_1.3.0              svglite_2.1.3            
[166] future_1.33.2             RANN_2.6.1                Cairo_1.6-2              
[169] spatstat.data_3.1-2       rstudioapi_0.16.0         cluster_2.1.6            
[172] spatstat.utils_3.0-5      hms_1.1.3                 fitdistrplus_1.2-1       
[175] munsell_0.5.1             cowplot_1.1.3             colorspace_2.1-0         
[178] FNN_1.1.4                 rlang_1.1.4               xts_0.14.0               
[181] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  dotCall64_1.1-1          
[184] shinydashboard_0.7.2      ggforce_0.4.2             circlize_0.4.16          
[187] laeken_0.5.3              mgcv_1.9-1                xfun_0.46                
[190] sna_2.7-2                 coda_0.19-4.1             e1071_1.7-14             
[193] TH.data_1.1-2             iterators_1.0.14          emmeans_1.10.3           
[196] abind_1.4-5               GOSemSim_2.30.0           treeio_1.28.0            
[199] bitops_1.0-7              Rdpack_2.6                promises_1.3.0           
[202] scatterpie_0.2.3          RSQLite_2.3.7             qvalue_2.36.0            
[205] sandwich_3.1-0            DelayedArray_0.30.1       proxy_0.4-27             
[208] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[211] boot_1.3-30               beachmat_2.20.0           listenv_0.9.1            
[214] Rcpp_1.0.13               edgeR_4.2.1               workflowr_1.7.1          
[217] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-61              
[220] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[223] spatstat.random_3.3-1     R6_2.5.1                  fastmap_1.2.0            
[226] multcomp_1.4-26           fastmatch_1.1-4           rstatix_0.7.2            
[229] TTR_0.24.4                vipor_0.4.7               ROCR_1.0-11              
[232] vcd_1.4-12                rsvd_1.0.5                nnet_7.3-19              
[235] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[238] deldir_2.0-4              ggthemes_5.1.0            htmltools_0.5.8.1        
[241] bit64_4.0.5               spatstat.explore_3.3-1    lifecycle_1.0.4          
[244] blme_1.0-5                nloptr_2.1.1              sass_0.4.9               
[247] vctrs_0.6.5               spatstat.geom_3.3-2       robustbase_0.99-3        
[250] NMF_0.27                  ggfun_0.1.5               future.apply_1.11.2      
[253] bslib_0.7.0               pillar_1.9.0              magick_2.8.4             
[256] pcaMethods_1.96.0         gplots_3.1.3.1            metapod_1.12.0           
[259] locfit_1.5-9.10           jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
