<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-06-23" />

<title>FIbroblasts D20 Naive</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CNS Fibroblast Immune Surveillance</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">FIbroblasts D20 Naive</h1>
<h4 class="date">2023-06-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CNS Fibroblast Immune
Surveillance/</code> <span class="glyphicon glyphicon-question-sign"
aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240724code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240724)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240724code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240724)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomvskipnessCNSFibroblastImmuneSurveillancetreed415bd4eb7d2750ae96f942127dc108bbfdbcc1etargetblankd415bd4a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/vskipness/CNS-Fibroblast-Immune-Surveillance/tree/d415bd4eb7d2750ae96f942127dc108bbfdbcc1e" target="_blank">d415bd4</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomvskipnessCNSFibroblastImmuneSurveillancetreed415bd4eb7d2750ae96f942127dc108bbfdbcc1etargetblankd415bd4a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/vskipness/CNS-Fibroblast-Immune-Surveillance/tree/d415bd4eb7d2750ae96f942127dc108bbfdbcc1e" target="_blank">d415bd4</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/T_cell_annotation.Rmd
    Untracked:  analysis/cellchat_EndC.Rmd
    Untracked:  analysis/cellchat_analysis.Rmd
    Untracked:  analysis/cluster_annotation.Rmd
    Untracked:  analysis/endothelial_cell_annotation.Rmd
    Untracked:  analysis/fibroblast_annotation.Rmd
    Untracked:  analysis/fibroblast_gene_expression_anaylsis.Rmd
    Untracked:  analysis/merge_and_QC.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>load packages</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(RColorBrewer)
  library(viridis)
  library(ggsci)
  library(scater)
  library(scran)
  library(pheatmap)
  library(biomaRt)
  library(fgsea)
  library(grid)
  library(gridExtra)
  library(msigdbr)
  library(muscat)
  library(ggpubr)
  library(ExploreSCdataSeurat3)
  library(enrichplot)
  #library(Matrix.utils)
  library(destiny)
  library(clusterProfiler)
  library(org.Mm.eg.db)
  library(DOSE)
  })</code></pre>
</div>
<div id="load-file" class="section level1">
<h1>load file</h1>
<pre class="r"><code>basedir &lt;- here()
seurat &lt;- readRDS(paste0(basedir, &quot;/data/fibroblasts.rds&quot;))</code></pre>
</div>
<div id="processing-file" class="section level1">
<h1>processing file</h1>
<pre class="r"><code>processing &lt;- function(seurat){
  seurat &lt;- NormalizeData(object = seurat)
  seurat &lt;- FindVariableFeatures(object = seurat)
  seurat &lt;- ScaleData(object = seurat, verbose = FALSE)

  seurat &lt;- RunPCA(object = seurat, npcs = 30, verbose = FALSE, )
  seurat &lt;- RunUMAP(object = seurat, reduction = &quot;pca&quot;, dims = 1:20)

  res = c(0.1, 0.2, 0.25, 0.4, 0.6, 0.8)
  seurat &lt;- FindNeighbors(object = seurat, reduction = &quot;pca&quot;, dims = 1:20)
  for(i in 1:length(res)){
    seurat &lt;- FindClusters(object = seurat, resolution = res[i], random.seed = 1234)
  }
  return(seurat)
}

seurat &lt;- processing(seurat)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 7997
Number of edges: 284010

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9733
Number of communities: 8
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 7997
Number of edges: 284010

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9584
Number of communities: 10
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 7997
Number of edges: 284010

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9511
Number of communities: 10
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 7997
Number of edges: 284010

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9315
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 7997
Number of edges: 284010

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9094
Number of communities: 14
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 7997
Number of edges: 284010

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8912
Number of communities: 17
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>Idents(seurat) &lt;- seurat$timepoint
seurat$timepoint &lt;- factor(seurat$timepoint, levels = c(&quot;d0&quot;, &quot;d8&quot;, &quot;d20&quot;))</code></pre>
</div>
<div id="different-timepoints" class="section level1">
<h1>different timepoints</h1>
<pre class="r"><code>coltimepoint &lt;- c((brewer.pal(11, &quot;PiYG&quot;)[c(9,3)]),
                (brewer.pal(11, &quot;PRGn&quot;)[c(3)]))
names(coltimepoint) &lt;- unique(seurat$timepoint)

DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;timepoint&quot;, pt.size = 0.4, cols = coltimepoint, shuffle = T)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/UMAP%20grouped%20by%20timepoints-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cell_count &lt;- data.frame(table(seurat$timepoint))
colnames(cell_count) &lt;- c(&quot;timepoint&quot;, &quot;count&quot;)
hsize &lt;- 1.5

ggplot(cell_count, aes(x = hsize, y = count, fill = timepoint)) +
  scale_fill_manual(values = coltimepoint) +
  geom_col(color = &quot;white&quot;) +
  coord_polar(theta = &quot;y&quot;) +
  xlim(c(0.2, hsize + 0.5)) +
  theme_void() +
  ggtitle(&quot;timepoint_count&quot;) +
  theme(plot.title = element_text(hjust = 0.5, size = 15)) +
  geom_text(aes(label = count), position = position_stack(vjust = 0.5))</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/cell%20count%20timepoint-1.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$RNA_snn_res.0.2
DimPlot(seurat, reduction = &quot;umap&quot;, split.by = &quot;timepoint&quot;, label = T, pt.size = 0.4)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/visualization%20split%20by%20timepoint%20res%200.2-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="fibroblast-annotation" class="section level1">
<h1>fibroblast annotation</h1>
<pre class="r"><code>Idents(seurat) &lt;- seurat$RNA_snn_res.0.2
DimPlot(seurat, reduction = &quot;umap&quot;, label = T, pt.size = 0.4)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/visualization%20fibroblast%20cluster%20res%200.2-1.png" width="672" style="display: block; margin: auto;" />
## run marker genes</p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterlabel

all.markers &lt;- FindAllMarkers(seurat, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25, test.use = &quot;wilcox&quot;)
all.markers %&gt;%
  group_by(cluster) %&gt;%
  top_n(n = 30, wt = avg_log2FC) -&gt; markers.top30
markers.top30</code></pre>
<pre><code># A tibble: 150 × 7
# Groups:   cluster [5]
   p_val avg_log2FC pct.1 pct.2 p_val_adj cluster       gene                      
   &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;         &lt;chr&gt;                     
 1     0       6.58 0.938 0.04          0 Fibroblasts 1 ENSMUSG00000027737.Slc7a11
 2     0       5.28 0.949 0.132         0 Fibroblasts 1 ENSMUSG00000039323.Igfbp2 
 3     0       4.99 0.914 0.108         0 Fibroblasts 1 ENSMUSG00000005360.Slc1a3 
 4     0       6.58 0.787 0.011         0 Fibroblasts 1 ENSMUSG00000013584.Aldh1a2
 5     0       5.64 0.714 0.035         0 Fibroblasts 1 ENSMUSG00000040055.Gjb6   
 6     0       5.41 0.718 0.042         0 Fibroblasts 1 ENSMUSG00000046352.Gjb2   
 7     0       9.31 0.671 0.004         0 Fibroblasts 1 ENSMUSG00000024650.Slc22a6
 8     0       7.54 0.658 0.008         0 Fibroblasts 1 ENSMUSG00000029843.Slc13a4
 9     0       5.20 0.695 0.048         0 Fibroblasts 1 ENSMUSG00000044349.Snhg11 
10     0       5.79 0.67  0.027         0 Fibroblasts 1 ENSMUSG00000039004.Bmp6   
# ℹ 140 more rows</code></pre>
<div id="assign-cluster-labels" class="section level2">
<h2>assign cluster labels</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$RNA_snn_res.0.2

seurat$clusterlabel &lt;- &quot;-&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;0&quot;)] &lt;- &quot;Pericytes&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;1&quot;)] &lt;- &quot;Periv. Fbs&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;2&quot;)] &lt;- &quot;Arachnoid Fbs&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;3&quot;)] &lt;- &quot;Pial Fbs&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;4&quot;)] &lt;- &quot;VSMCs&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;5&quot;)] &lt;- &quot;?&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;6&quot;)] &lt;- &quot;EndC&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;7&quot;)] &lt;- &quot;Periv. Fbs activated&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;8&quot;)] &lt;- &quot;CP/ME periv. Fbs&quot;
seurat$clusterlabel[which(Idents(seurat)==&quot;9&quot;)] &lt;- &quot;Dural Fbs&quot;
Idents(seurat) &lt;- seurat$clusterlabel

seurat$clusterlabel &lt;- factor(seurat$clusterlabel, levels = c(&quot;EndC&quot;, &quot;Pericytes&quot;, &quot;Periv. Fbs&quot;, &quot;Periv. Fbs activated&quot;, &quot;VSMCs&quot;, &quot;Pial Fbs&quot;, &quot;Arachnoid Fbs&quot;, &quot;Dural Fbs&quot;, &quot;CP/ME periv. Fbs&quot;, &quot;?&quot;))

Idents(seurat) &lt;- seurat$clusterlabel
DimPlot(seurat, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/assign%20labels-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="removal-of-endothelial-cells" class="section level2">
<h2>removal of endothelial cells</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterlabel
seurat &lt;- subset(seurat, idents = &quot;EndC&quot;, invert = T)

## color palette clusters
library(pals)
colcluster &lt;- c(tol()[c(1:9)])
names(colcluster) &lt;- c(&quot;Pericytes&quot;, &quot;Periv. Fbs&quot;, &quot;Periv. Fbs activated&quot;, &quot;VSMCs&quot;, &quot;Pial Fbs&quot;, &quot;Arachnoid Fbs&quot;, &quot;Dural Fbs&quot;, &quot;CP/ME periv. Fbs&quot;, &quot;?&quot;)
pal.bands(colcluster)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/removal%20of%20endothelial%20cells-1.png" width="691.2" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterlabel
DimPlot(seurat, reduction = &quot;umap&quot;, cols = colcluster)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/removal%20of%20endothelial%20cells-2.png" width="691.2" style="display: block; margin: auto;" /></p>
</div>
<div id="timpoint-visualizations-without-endc-cluster"
class="section level2">
<h2>timpoint visualizations without EndC cluster</h2>
<pre class="r"><code>coltimepoint &lt;- c((brewer.pal(11, &quot;PiYG&quot;)[c(9,3)]),
                (brewer.pal(11, &quot;PRGn&quot;)[c(3)]))
names(coltimepoint) &lt;- unique(seurat$timepoint)

DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;timepoint&quot;, pt.size = 0.4, cols = coltimepoint, shuffle = T)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/UMAP%20grouped%20by%20timepoints%20w/o%20EndC-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cell_count &lt;- data.frame(table(seurat$timepoint))
colnames(cell_count) &lt;- c(&quot;timepoint&quot;, &quot;count&quot;)
hsize &lt;- 1.5

ggplot(cell_count, aes(x = hsize, y = count, fill = timepoint)) +
  scale_fill_manual(values = coltimepoint) +
  geom_col(color = &quot;white&quot;) +
  coord_polar(theta = &quot;y&quot;) +
  xlim(c(0.2, hsize + 0.5)) +
  theme_void() +
  ggtitle(&quot;timepoint_count&quot;) +
  theme(plot.title = element_text(hjust = 0.5, size = 15)) +
  geom_text(aes(label = count), position = position_stack(vjust = 0.5))</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/cell%20count%20timepoint%20w/o%20EndC-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="dotplot-annotation" class="section level2">
<h2>dotplot annotation</h2>
<pre class="r"><code>genes &lt;- data.frame(gene = rownames(seurat)) %&gt;%
  mutate(geneID = gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID =c(&quot;Kcnj8&quot;, &quot;Pdgfrb&quot;, &quot;Rgs5&quot;, &quot;Myl9&quot;, &quot;Notch3&quot;, &quot;Mcam&quot;, &quot;Myh11&quot;, &quot;Tagln&quot;, &quot;Acta2&quot;, &quot;Des&quot;, &quot;Il34&quot;, &quot;Ccl19&quot;, &quot;Col3a1&quot;, &quot;Cxcl9&quot;, &quot;Cxcl10&quot;, &quot;Bgn&quot;, &quot;Dcn&quot;, &quot;Col1a1&quot;, &quot;Lum&quot;, &quot;Pdgfra&quot;, &quot;S100a6&quot;, &quot;Lama1&quot;, &quot;Col15a1&quot;, &quot;Col12a1&quot;, &quot;Slc6a13&quot;, &quot;Spp1&quot;, &quot;Edn3&quot;, &quot;Vwa1&quot;, &quot;Alcam&quot;, &quot;Aldh1a2&quot;, &quot;Ptgds&quot;, &quot;Pdpn&quot;, &quot;Slc13a3&quot;, &quot;Crabp2&quot;, &quot;Cxcl16&quot;, &quot;Cygb&quot;, &quot;Cd34&quot;, &quot;Padi4&quot;, &quot;Ltbp4&quot;, &quot;Abi3bp&quot;, &quot;Cntfr&quot;, &quot;Tcf21&quot;, &quot;Alpl&quot;, &quot;Fxyd5&quot;, &quot;Slc16a1&quot;, &quot;Ogn&quot;))%&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

#selGenes &lt;- data.frame(geneID =c(&quot;Cxcl16&quot;, &quot;Dcn&quot;, &quot;Vcam1&quot;, &quot;Col1a1&quot;, &quot;Lum&quot;, &quot;Pdgfra&quot;, &quot;Bgn&quot;, &quot;Rgs5&quot;, &quot;Pdgfrb&quot;, &quot;Ccl19&quot;, &quot;Myl9&quot;, &quot;Notch3&quot;, &quot;Tagln&quot;, &quot;Tuba1b&quot;, &quot;Plp1&quot;, &quot;S100b&quot;, &quot;Cldn11&quot;, &quot;Acta2&quot;, &quot;Myh11&quot;, &quot;Des&quot;, &quot;Mcam&quot;, &quot;Pdpn&quot;, &quot;Plpp3&quot;, &quot;Kcnj8&quot;, &quot;Pecam1&quot;, &quot;Lama1&quot;, &quot;Col15a1&quot;, &quot;Dpp4&quot;, &quot;Slc47a1&quot;, &quot;Ppp1r1a&quot;, &quot;Slc7a11&quot;, &quot;Tcf21&quot;, &quot;Vwa1&quot;, &quot;Cdh5&quot;, &quot;Prox1&quot;)) %&gt;%
#  left_join(., genes, by = &quot;geneID&quot;)

#selGenes &lt;- data.frame(geneID =c(&quot;Dcn&quot;, &quot;Vcam1&quot;, &quot;Col1a1&quot;, &quot;Lum&quot;, &quot;Pdgfra&quot;, &quot;Bgn&quot;, &quot;Rgs5&quot;, &quot;Pdgfrb&quot;, &quot;Myl9&quot;, &quot;Notch3&quot;, &quot;Tagln&quot;, &quot;Tuba1b&quot;, &quot;Plp1&quot;, &quot;S100b&quot;, &quot;Cldn11&quot;, &quot;Acta2&quot;, &quot;Myh11&quot;, &quot;Des&quot;, &quot;Plpp3&quot;, &quot;Kcnj8&quot;, &quot;Pecam1&quot;, &quot;Lama1&quot;, &quot;Lama2&quot;, &quot;Col15a1&quot;, &quot;Dpp4&quot;, &quot;Slc47a1&quot;, &quot;Ppp1r1a&quot;, &quot;Slc7a11&quot;, &quot;Tcf21&quot;, &quot;Vwa1&quot;, &quot;Cdh5&quot;, &quot;Prox1&quot;, &quot;Gfap&quot;, &quot;Cxcl10&quot;, &quot;Cxcl9&quot;, &quot;Cxcl16&quot;, &quot;Ccl19&quot;, &quot;Rosa26eyfp&quot;, &quot;Cspg4&quot;, &quot;Mcam&quot;, &quot;Il33&quot;, &quot;Il34&quot;, &quot;Cx30&quot;, &quot;Pdpn&quot;)) %&gt;%
#  left_join(., genes, by = &quot;geneID&quot;)

#periv 
#selGenes &lt;- data.frame(geneID =c(&quot;Col15a1&quot;, &quot;Lama1&quot;, &quot;Vtn&quot;, &quot;Flrt2&quot;, &quot;Slc6a13&quot;, &quot;Slc30a10&quot;, &quot;Fbln2&quot;, &quot;Cdh5&quot;)) %&gt;%
#  left_join(., genes, by = &quot;geneID&quot;)

#periv CE/ME
#selGenes &lt;- data.frame(geneID =c(&quot;Tcf21&quot;, &quot;Mylk&quot;, &quot;Tbc1d2b&quot;, &quot;Tmem176a&quot;, &quot;Dach1&quot;, &quot;Col26a1&quot;)) %&gt;%
#  left_join(., genes, by = &quot;geneID&quot;)

#arachnoid
#selGenes &lt;- data.frame(geneID =c(&quot;Dpp4&quot;, &quot;Ppp1r1a&quot;, &quot;Slc7a11&quot;, &quot;Cdh5&quot;, &quot;Cdh1&quot;, &quot;Cldn11&quot;, &quot;Prox1&quot;)) %&gt;%
#  left_join(., genes, by = &quot;geneID&quot;)

#pia
#selGenes &lt;- data.frame(geneID =c(&quot;Lama1&quot;, &quot;Col15a1&quot;, &quot;Cdh5&quot;, &quot;Vtn&quot;)) %&gt;%
#  left_join(., genes, by = &quot;geneID&quot;)

#dural
#selGenes &lt;- data.frame(geneID =c(&quot;Slc47a1&quot;, &quot;Col1a1&quot;, &quot;Pcdh17&quot;, &quot;Calca&quot;, &quot;Cdh5&quot;, &quot;Cdh11&quot;)) %&gt;%
#  left_join(., genes, by = &quot;geneID&quot;)

Idents(seurat) &lt;- seurat$clusterlabel

DotPlot(seurat, assay = &quot;RNA&quot;, features = rev(selGenes$gene), scale = T, cluster.idents = F) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_x_discrete(breaks = rev(selGenes$gene), labels = rev(selGenes$geneID)) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/dotplot%20annotation-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="finding-marker-genes-for-ecm-producing-fibroblasts"
class="section level2">
<h2>finding marker genes for ECM producing fibroblasts</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterlabel
levels(seurat)</code></pre>
<pre><code>[1] &quot;Pericytes&quot;            &quot;Periv. Fbs&quot;           &quot;Periv. Fbs activated&quot; &quot;VSMCs&quot;               
[5] &quot;Pial Fbs&quot;             &quot;Arachnoid Fbs&quot;        &quot;Dural Fbs&quot;            &quot;CP/ME periv. Fbs&quot;    
[9] &quot;?&quot;                   </code></pre>
<pre class="r"><code>#all.markers &lt;- FindMarkers(seurat, ident.1 = &quot;?&quot;, ident.2 = c(&quot;Pial fibroblasts&quot;, &quot;Arachnoid fibroblasts&quot;, &quot;CP/ME periv fibroblasts&quot;, &quot;Dural fibroblasts&quot;), min.pct = 0.25, logfc.threshold = 0.25, test.use = &quot;wilcox&quot;, only.pos = T)
#all.markers %&gt;%
#  top_n(n = 40, wt = avg_log2FC) -&gt; markers.top20
#markers.top20</code></pre>
</div>
<div id="frequencies-per-timepoint" class="section level2">
<h2>frequencies per timepoint</h2>
<pre class="r"><code>datList &lt;- NULL
for(t in unique(seurat$timepoint)){
  seuratSub &lt;- subset(seurat, timepoint==t)
  print(dim(seuratSub))
  dat_t &lt;- as.data.frame(table(seuratSub$clusterlabel)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(timepoint=t)
  datList[[t]] &lt;- dat_t
}</code></pre>
<pre><code>[1] 30150  3590
[1] 30150  1191
[1] 30150  2976</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

## plot abundance
ggbarplot(dat_all, x= &quot;timepoint&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;, palette = colcluster)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))  # scale_x_discrete(limits=orddatasets)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/frequencies%20per%20timepoint-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="activation-of-perivascular-fibroblasts" class="section level1">
<h1>activation of perivascular fibroblasts</h1>
<pre class="r"><code>seurat_sub &lt;- subset(seurat, idents = c(&quot;Periv. Fbs&quot;, &quot;Periv. Fbs activated&quot;))
Idents(seurat) &lt;- seurat_sub$clusterlabel


selGenes &lt;- data.frame(geneID = c(&quot;Ccl19&quot;, &quot;Cxcl10&quot;, &quot;Cxcl9&quot;)) %&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

for (i in 1:length(selGenes$gene)){
  print(VlnPlot(seurat_sub, features = selGenes$gene[i], pt.size = 0, cols = colcluster)+
          theme_classic() +
          ggtitle(selGenes$geneID[i]) +
          theme(plot.title = element_text(hjust = 0.5, size = 25), legend.position = &quot;none&quot;) + labs(x = &quot;&quot;))
}</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/activation%20of%20perivascular%20fibroblasts%20vln%20plot-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/fibroblast_annotation.Rmd/activation%20of%20perivascular%20fibroblasts%20vln%20plot-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/fibroblast_annotation.Rmd/activation%20of%20perivascular%20fibroblasts%20vln%20plot-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>datList &lt;- NULL
for(t in unique(seurat_sub$timepoint)){
  seuratSub &lt;- subset(seurat_sub, timepoint==t)
  print(dim(seuratSub))
  dat_t &lt;- as.data.frame(table(seuratSub$clusterlabel)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(timepoint=t)
  datList[[t]] &lt;- dat_t
}</code></pre>
<pre><code>[1] 30150   688
[1] 30150   243
[1] 30150   929</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

## plot abundance
ggbarplot(dat_all, x= &quot;timepoint&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;, palette = colcluster)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))  # scale_x_discrete(limits=orddatasets)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/activation%20of%20perivascular%20fibroblasts%20box%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="look-at-rosa26eyfp-expression" class="section level1">
<h1>look at Rosa26eyfp expression</h1>
<pre class="r"><code>genes &lt;- data.frame(gene = rownames(seurat)) %&gt;%
  mutate(geneID = gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))
selGenes &lt;- data.frame(geneID = c(&quot;Ccl19&quot;, &quot;Rosa26eyfp&quot;, &quot;Cxcl16&quot;)) %&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

FeaturePlot(object = seurat, features = selGenes$gene, pt.size = 1, cols = c(&quot;gray&quot;,&quot;magenta3&quot;), split.by = &quot;timepoint&quot;)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/Featureplot%20Ccl19%20Rosa26eyfp%20and%20Cxcl16-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#genes %&gt;% filter(grepl(&quot;Rosa&quot;, gene))</code></pre>
<div id="removal-of-b6-cells" class="section level2">
<h2>removal of B6 cells</h2>
<pre class="r"><code># Subset on the expression level of a gene/feature
Idents(seurat) &lt;- seurat$dataset
seurat_sub &lt;- subset(seurat, idents = c(&quot;316781_01-1_20230323_Mu_B6_CNS_Naive_Stroma_1&quot;, &quot;316781_02-2_20230323_Mu_B6_CNS_Naive_Stroma_2&quot;), invert = T)

genes &lt;- data.frame(gene = rownames(seurat_sub)) %&gt;%
  mutate(geneID = gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))
selGenes &lt;- data.frame(geneID = c(&quot;Ccl19&quot;, &quot;Rosa26eyfp&quot;, &quot;Cxcl16&quot;)) %&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

FeaturePlot(object = seurat_sub, features = selGenes$gene, pt.size = 1, cols = c(&quot;gray&quot;,&quot;magenta3&quot;), split.by = &quot;timepoint&quot;)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/Featureplot%20Ccl19%20Rosa26eyfp%20and%20Cxcl16%20after%20removal%20of%20B6%20cells-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#genes %&gt;% filter(grepl(&quot;Rosa&quot;, gene))</code></pre>
</div>
<div id="rosa26eyfp-expression-frequency" class="section level2">
<h2>Rosa26eyfp expression frequency</h2>
<pre class="r"><code>Rosa26 &lt;- subset(seurat_sub, subset = Rosa26eyfp.Rosa26eyfp &gt; 0)
No_Rosa26 &lt;- subset(seurat_sub, subset = Rosa26eyfp.Rosa26eyfp &gt; 0, invert = T)
Rosa26$Rosa26 &lt;- &quot;yes&quot;
No_Rosa26$Rosa26 &lt;- &quot;no&quot;
seurat_sub &lt;- merge(No_Rosa26, Rosa26)
table(seurat_sub$Rosa26, seurat_sub$clusterlabel)</code></pre>
<pre><code>     
         ? Arachnoid Fbs CP/ME periv. Fbs Dural Fbs Pericytes Periv. Fbs Periv. Fbs activated
  no   177           739               87        82       530       1045                  153
  yes  282           684               80        61        26        528                   71
     
      Pial Fbs VSMCs
  no       788   460
  yes      333   155</code></pre>
<pre class="r"><code>rm(Rosa26, No_Rosa26)

datList &lt;- NULL
for(c in unique(seurat_sub$clusterlabel)){
  seuratSub &lt;- subset(seurat_sub, clusterlabel==c)
  print(dim(seuratSub))
  dat_c &lt;- as.data.frame(table(seuratSub$Rosa26)) %&gt;% mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(clusterlabel=c)
  datList[[c]] &lt;- dat_c
}</code></pre>
<pre><code>[1] 30150   459
[1] 30150   615
[1] 30150  1121
[1] 30150  1573
[1] 30150   556
[1] 30150  1423
[1] 30150   167
[1] 30150   143
[1] 30150   224</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

colRosa &lt;- brewer.pal(11, &quot;PiYG&quot;)[c(8,4)]
names(colRosa) &lt;- c(&quot;yes&quot;, &quot;no&quot;)

## plot abundance
ggbarplot(dat_all, x= &quot;clusterlabel&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;, palette = colRosa)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))  # scale_x_discrete(limits=orddatasets)</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/Rosa26eyfp%20expression%20frequency%20vln%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ccl19-and-cxcl16-expression-of-rosa26eyfp--cells"
class="section level2">
<h2>Ccl19 and Cxcl16 expression of Rosa26eyfp+/- cells</h2>
<pre class="r"><code>Idents(seurat_sub) &lt;- seurat_sub$clusterlabel
VlnPlot(seurat_sub, features = c(&quot;ENSMUSG00000071005.Ccl19&quot;, &quot;Rosa26eyfp.Rosa26eyfp&quot;, &quot;ENSMUSG00000018920.Cxcl16&quot;), split.by = &quot;Rosa26&quot;, cols = rev(colRosa))</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/violin%20plot-1.png" width="1920" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## color names somehow not taken into account here therefore reversed colors to make it right</code></pre>
</div>
</div>
<div id="feature-plots-of-ccl19-and-cxcl16" class="section level1">
<h1>feature plots of Ccl19 and Cxcl16</h1>
<pre class="r"><code>selGenes &lt;- data.frame(geneID = c(&quot;Ccl19&quot;, &quot;Cxcl16&quot;)) %&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

for (i in 1:length(selGenes$gene)){
  print(FeaturePlot(seurat, features = selGenes$gene[i], cols = c(&quot;grey&quot;, &quot;magenta3&quot;), pt.size = 1)+
          theme_void() +
          labs(title = selGenes$geneID[i]) +
          ggtitle(selGenes$geneID[i]) +
          theme(plot.title = element_text(hjust = 0.5, size = 25)))
}</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/feature%20plot%20Ccl19%20and%20Cxcl16%20combined-1.png" width="528" style="display: block; margin: auto;" /><img src="figure/fibroblast_annotation.Rmd/feature%20plot%20Ccl19%20and%20Cxcl16%20combined-2.png" width="528" style="display: block; margin: auto;" /></p>
<div id="d0" class="section level2">
<h2>d0</h2>
<pre class="r"><code>selGenes &lt;- data.frame(geneID = c(&quot;Ccl19&quot;, &quot;Cxcl16&quot;)) %&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

seurat_sub &lt;- subset(seurat, timepoint %in% &quot;d0&quot;)

for (i in 1:length(selGenes$gene)){
  print(FeaturePlot(seurat_sub, features = selGenes$gene[i], cols = c(&quot;grey&quot;, &quot;magenta3&quot;), pt.size = 1)+
          theme_void() +
          labs(title = selGenes$geneID[i]) +
          ggtitle(selGenes$geneID[i]) +
          theme(plot.title = element_text(hjust = 0.5, size = 25)))
}</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/feature%20plot%20Ccl19%20and%20Cxcl16%20d0-1.png" width="528" style="display: block; margin: auto;" /><img src="figure/fibroblast_annotation.Rmd/feature%20plot%20Ccl19%20and%20Cxcl16%20d0-2.png" width="528" style="display: block; margin: auto;" /></p>
</div>
<div id="d8" class="section level2">
<h2>d8</h2>
<pre class="r"><code>selGenes &lt;- data.frame(geneID = c(&quot;Ccl19&quot;, &quot;Cxcl16&quot;)) %&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

seurat_sub &lt;- subset(seurat, timepoint %in% &quot;d8&quot;)

for (i in 1:length(selGenes$gene)){
  print(FeaturePlot(seurat_sub, features = selGenes$gene[i], cols = c(&quot;grey&quot;, &quot;magenta3&quot;), pt.size = 1)+
          theme_void() +
          labs(title = selGenes$geneID[i]) +
          ggtitle(selGenes$geneID[i]) +
          theme(plot.title = element_text(hjust = 0.5, size = 25)))
}</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/feature%20plot%20Ccl19%20and%20Cxcl16%20d8-1.png" width="528" style="display: block; margin: auto;" /><img src="figure/fibroblast_annotation.Rmd/feature%20plot%20Ccl19%20and%20Cxcl16%20d8-2.png" width="528" style="display: block; margin: auto;" /></p>
</div>
<div id="d20" class="section level2">
<h2>d20</h2>
<pre class="r"><code>selGenes &lt;- data.frame(geneID = c(&quot;Ccl19&quot;, &quot;Cxcl16&quot;)) %&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

seurat_sub &lt;- subset(seurat, timepoint %in% &quot;d20&quot;)

for (i in 1:length(selGenes$gene)){
  print(FeaturePlot(seurat_sub, features = selGenes$gene[i], cols = c(&quot;grey&quot;, &quot;magenta3&quot;), pt.size = 1)+
          theme_void() +
          labs(title = selGenes$geneID[i]) +
          ggtitle(selGenes$geneID[i]) +
          theme(plot.title = element_text(hjust = 0.5, size = 25)))
}</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/feature%20plot%20Ccl19%20and%20Cxcl16%20d20-1.png" width="528" style="display: block; margin: auto;" /><img src="figure/fibroblast_annotation.Rmd/feature%20plot%20Ccl19%20and%20Cxcl16%20d20-2.png" width="528" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="violin-plots-ccl19-and-cxcl16" class="section level1">
<h1>violin plots Ccl19 and Cxcl16</h1>
<pre class="r"><code>selGenes &lt;- data.frame(geneID = c(&quot;Ccl19&quot;, &quot;Cxcl16&quot;)) %&gt;%
  left_join(., genes, by = &quot;geneID&quot;)

Idents(seurat) &lt;- seurat$timepoint

for (i in 1:length(selGenes$gene)){
  print(VlnPlot(seurat, features = selGenes$gene[i], pt.size = 0, cols = coltimepoint)+
          theme_classic() +
          ggtitle(selGenes$geneID[i]) +
          theme(plot.title = element_text(hjust = 0.5, size = 25), legend.position = &quot;none&quot;) + labs(x = &quot;&quot;))
}</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/violin%20plots%20Ccl19%20and%20Cxcl16%20expression%20at%20diff.%20timepoints-1.png" width="480" style="display: block; margin: auto;" /><img src="figure/fibroblast_annotation.Rmd/violin%20plots%20Ccl19%20and%20Cxcl16%20expression%20at%20diff.%20timepoints-2.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterlabel

for (i in 1:length(selGenes$gene)){
  print(VlnPlot(seurat, features = selGenes$gene[i], pt.size = 0, split.by = &quot;timepoint&quot;, cols = coltimepoint)+
          theme_classic() +
          ggtitle(selGenes$geneID[i]) +
          theme(plot.title = element_text(hjust = 0.5, size = 25))+ labs(x = &quot;&quot;))
}</code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/violin%20plots%20Ccl19%20and%20Cxcl16%20diff.%20cluster%20split%20by%20timepoint-1.png" width="960" style="display: block; margin: auto;" /><img src="figure/fibroblast_annotation.Rmd/violin%20plots%20Ccl19%20and%20Cxcl16%20diff.%20cluster%20split%20by%20timepoint-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="project-genes-on-umap" class="section level1">
<h1>project genes on umap</h1>
<pre class="r"><code>##convert seurat object to sce object

sce &lt;- as.SingleCellExperiment(seurat)

##define genes to project

Meningeal &lt;- data.frame(gene=c(&quot;Col1a1&quot;, &quot;Lum&quot;, &quot;Dcn&quot;, &quot;Pdgfra&quot;))
Perivascular &lt;- data.frame(gene=c(&quot;Ccl19&quot;, &quot;Il34&quot;, &quot;Col3a1&quot;))

genes &lt;- data.frame(geneID=rownames(sce)) %&gt;% mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;, geneID))

## color palette
pal = colorRampPalette(rev(brewer.pal(11, &quot;RdBu&quot;)))
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 5))
sf &lt;- scale_fill_gradientn(colours = pal(100), limits=c(0, 5))
 
##make a count matrix of signature genes for ECM

signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% Meningeal$gene)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
 
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; 1.5)] &lt;- 1.5
plotUMAP(sceSub, colour_by = &quot;sign&quot;, point_size = 1) + sf + 
  theme_void() + 
  ggtitle(&quot;Meningeal Fbs&quot;, subtitle = &quot;Col1a1, Lum, Vcam1, Pdpn, Col15a1&quot;) + 
  theme(plot.title = element_text(hjust = 0.5, size = 15), 
        plot.subtitle = element_text(hjust = 0.5, face = &quot;italic&quot;)) </code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/project%20gene%20signatures-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code> ##make a count matrix of signature genes for Ccl19

signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% Perivascular$gene)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
 
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; 1.5)] &lt;- 1.5

plotUMAP(sceSub, colour_by = &quot;sign&quot;, point_size = 1) + sf + 
  theme_void() + 
  ggtitle(&quot;Perivascular Fbs&quot;, subtitle = &quot;Ccl19, Il34, Col3a1&quot;) + 
  theme(plot.title = element_text(hjust = 0.5, size = 15), 
        plot.subtitle = element_text(hjust = 0.5, face = &quot;italic&quot;)) </code></pre>
<p><img src="figure/fibroblast_annotation.Rmd/project%20gene%20signatures-2.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="save-file" class="section level1">
<h1>save file</h1>
<pre class="r"><code>saveRDS(seurat, file = paste0(basedir, &quot;/data/fibroblasts_annotated.rds&quot;))</code></pre>
</div>
<div id="sessioninfo" class="section level1">
<h1>sessionInfo</h1>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-apple-darwin20
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pals_1.9                    DOSE_3.30.2                 org.Mm.eg.db_3.19.1        
 [4] AnnotationDbi_1.66.0        clusterProfiler_4.12.1      destiny_3.18.0             
 [7] enrichplot_1.24.2           ExploreSCdataSeurat3_0.1.0  ggpubr_0.6.0               
[10] muscat_1.18.0               msigdbr_7.5.1               gridExtra_2.3              
[13] fgsea_1.30.0                biomaRt_2.60.1              pheatmap_1.0.12            
[16] scran_1.32.0                scater_1.32.1               scuttle_1.14.0             
[19] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[22] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[25] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[28] matrixStats_1.3.0           ggsci_3.2.0                 viridis_0.6.5              
[31] viridisLite_0.4.2           RColorBrewer_1.1-3          here_1.0.1                 
[34] magrittr_2.0.3              Seurat_5.1.0                SeuratObject_5.0.2         
[37] sp_2.1-4                    lubridate_1.9.3             forcats_1.0.0              
[40] stringr_1.5.1               dplyr_1.1.4                 purrr_1.0.2                
[43] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[46] ggplot2_3.5.1               tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] igraph_2.0.3              ica_1.0-3                 plotly_4.10.4            
  [4] maps_3.4.2                zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.0.5                 doParallel_1.0.17         clue_0.3-65              
 [10] lattice_0.22-6            rjson_0.2.21              blob_1.2.4               
 [13] S4Arrays_1.4.1            dichromat_2.0-0.1         pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 cli_3.6.3                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  bluster_1.14.0            BiocNeighbors_1.22.0     
 [25] uwot_0.2.2                shadowtext_0.1.4          curl_5.2.1               
 [28] mime_0.12                 evaluate_0.24.0           tidytree_0.4.6           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           lmerTest_3.1-3            httpuv_1.6.15            
 [37] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [40] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] smoother_1.3              jquerylib_0.1.4           withr_3.0.0              
 [46] git2r_0.33.0              corpcor_1.6.10            class_7.3-22             
 [49] rprojroot_2.0.4           lmtest_0.9-40             tidygraph_1.3.1          
 [52] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [55] ggrepel_0.9.5             labeling_0.4.3            fANCOVA_0.6-1            
 [58] SparseArray_1.4.8         ranger_0.16.0             DESeq2_1.44.0            
 [61] DEoptimR_1.1-3            reticulate_1.38.0         hexbin_1.28.3            
 [64] zoo_1.8-12                XVector_0.44.0            knitr_1.48               
 [67] ggplot.multistats_1.0.0   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [70] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [73] patchwork_1.2.0           caTools_1.18.2            data.table_1.15.4        
 [76] ggtree_3.12.0             RSpectra_0.16-2           irlba_2.3.5.1            
 [79] ggrastr_1.0.2             fastDummies_1.7.3         gridGraphics_0.5-1       
 [82] lazyeval_0.2.2            yaml_2.3.9                survival_3.7-0           
 [85] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [88] mapproj_1.2.11            progressr_0.14.0          tweenr_2.0.3             
 [91] later_1.3.2               ggridges_0.5.6            codetools_0.2-20         
 [94] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
 [97] Rtsne_0.17                shape_1.4.6.1             limma_3.60.4             
[100] estimability_1.5.1        filelock_1.0.3            pkgconfig_2.0.3          
[103] xml2_1.3.6                TMB_1.9.14                spatstat.univar_3.0-0    
[106] EnvStats_2.8.1            scatterplot3d_0.3-44      aplot_0.2.3              
[109] spatstat.sparse_3.1-0     ape_5.8                   xtable_1.8-4             
[112] highr_0.11                car_3.1-2                 plyr_1.8.9               
[115] httr_1.4.7                rbibutils_2.2.16          tools_4.4.1              
[118] globals_0.16.3            beeswarm_0.4.0            broom_1.0.6              
[121] nlme_3.1-165              HDO.db_0.99.1             dbplyr_2.5.0             
[124] lme4_1.1-35.5             digest_0.6.36             numDeriv_2016.8-1.1      
[127] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[130] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.1.5        
[133] glue_1.7.0                cachem_1.1.0              BiocFileCache_2.12.0     
[136] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[139] mvtnorm_1.2-5             presto_1.0.0              parallelly_1.37.1        
[142] statmod_1.5.0             RcppHNSW_0.6.0            ScaledMatrix_1.12.0      
[145] carData_3.0-5             minqa_1.2.7               pbapply_1.7-2            
[148] httr2_1.0.2               spam_2.10-0               gson_0.1.0               
[151] dqrng_0.4.1               utf8_1.2.4                graphlayouts_1.1.1       
[154] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.0      
[157] shiny_1.8.1.1             GenomeInfoDbData_1.2.12   glmmTMB_1.1.9            
[160] memoise_2.0.1             rmarkdown_2.27            scales_1.3.0             
[163] future_1.33.2             RANN_2.6.1                Cairo_1.6-2              
[166] spatstat.data_3.1-2       rstudioapi_0.16.0         cluster_2.1.6            
[169] spatstat.utils_3.0-5      hms_1.1.3                 fitdistrplus_1.2-1       
[172] munsell_0.5.1             cowplot_1.1.3             colorspace_2.1-0         
[175] rlang_1.1.4               xts_0.14.0                DelayedMatrixStats_1.26.0
[178] sparseMatrixStats_1.16.0  dotCall64_1.1-1           shinydashboard_0.7.2     
[181] ggforce_0.4.2             circlize_0.4.16           laeken_0.5.3             
[184] mgcv_1.9-1                xfun_0.46                 coda_0.19-4.1            
[187] e1071_1.7-14              TH.data_1.1-2             iterators_1.0.14         
[190] emmeans_1.10.3            abind_1.4-5               GOSemSim_2.30.0          
[193] treeio_1.28.0             bitops_1.0-7              Rdpack_2.6               
[196] promises_1.3.0            scatterpie_0.2.3          RSQLite_2.3.7            
[199] qvalue_2.36.0             sandwich_3.1-0            DelayedArray_0.30.1      
[202] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.1           
[205] prettyunits_1.2.0         boot_1.3-30               beachmat_2.20.0          
[208] listenv_0.9.1             Rcpp_1.0.13               edgeR_4.2.1              
[211] workflowr_1.7.1           BiocSingular_1.20.0       tensor_1.5               
[214] MASS_7.3-61               progress_1.2.3            BiocParallel_1.38.0      
[217] babelgene_22.9            spatstat.random_3.3-1     R6_2.5.1                 
[220] fastmap_1.2.0             multcomp_1.4-26           fastmatch_1.1-4          
[223] rstatix_0.7.2             TTR_0.24.4                vipor_0.4.7              
[226] ROCR_1.0-11               vcd_1.4-12                rsvd_1.0.5               
[229] nnet_7.3-19               gtable_0.3.5              KernSmooth_2.23-24       
[232] miniUI_0.1.1.1            deldir_2.0-4              ggthemes_5.1.0           
[235] htmltools_0.5.8.1         bit64_4.0.5               spatstat.explore_3.3-1   
[238] lifecycle_1.0.4           blme_1.0-5                nloptr_2.1.1             
[241] sass_0.4.9                vctrs_0.6.5               spatstat.geom_3.3-2      
[244] robustbase_0.99-3         ggfun_0.1.5               future.apply_1.11.2      
[247] bslib_0.7.0               pillar_1.9.0              pcaMethods_1.96.0        
[250] gplots_3.1.3.1            metapod_1.12.0            locfit_1.5-9.10          
[253] jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Fri Aug  9 16:39:53 2024&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-apple-darwin20
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pals_1.9                    DOSE_3.30.2                 org.Mm.eg.db_3.19.1        
 [4] AnnotationDbi_1.66.0        clusterProfiler_4.12.1      destiny_3.18.0             
 [7] enrichplot_1.24.2           ExploreSCdataSeurat3_0.1.0  ggpubr_0.6.0               
[10] muscat_1.18.0               msigdbr_7.5.1               gridExtra_2.3              
[13] fgsea_1.30.0                biomaRt_2.60.1              pheatmap_1.0.12            
[16] scran_1.32.0                scater_1.32.1               scuttle_1.14.0             
[19] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[22] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[25] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[28] matrixStats_1.3.0           ggsci_3.2.0                 viridis_0.6.5              
[31] viridisLite_0.4.2           RColorBrewer_1.1-3          here_1.0.1                 
[34] magrittr_2.0.3              Seurat_5.1.0                SeuratObject_5.0.2         
[37] sp_2.1-4                    lubridate_1.9.3             forcats_1.0.0              
[40] stringr_1.5.1               dplyr_1.1.4                 purrr_1.0.2                
[43] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[46] ggplot2_3.5.1               tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] igraph_2.0.3              ica_1.0-3                 plotly_4.10.4            
  [4] maps_3.4.2                zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.0.5                 doParallel_1.0.17         clue_0.3-65              
 [10] lattice_0.22-6            rjson_0.2.21              blob_1.2.4               
 [13] S4Arrays_1.4.1            dichromat_2.0-0.1         pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 cli_3.6.3                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  bluster_1.14.0            BiocNeighbors_1.22.0     
 [25] uwot_0.2.2                shadowtext_0.1.4          curl_5.2.1               
 [28] mime_0.12                 evaluate_0.24.0           tidytree_0.4.6           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           lmerTest_3.1-3            httpuv_1.6.15            
 [37] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [40] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] smoother_1.3              jquerylib_0.1.4           withr_3.0.0              
 [46] git2r_0.33.0              corpcor_1.6.10            class_7.3-22             
 [49] rprojroot_2.0.4           lmtest_0.9-40             tidygraph_1.3.1          
 [52] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [55] ggrepel_0.9.5             labeling_0.4.3            fANCOVA_0.6-1            
 [58] SparseArray_1.4.8         ranger_0.16.0             DESeq2_1.44.0            
 [61] DEoptimR_1.1-3            reticulate_1.38.0         hexbin_1.28.3            
 [64] zoo_1.8-12                XVector_0.44.0            knitr_1.48               
 [67] ggplot.multistats_1.0.0   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [70] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [73] patchwork_1.2.0           caTools_1.18.2            data.table_1.15.4        
 [76] ggtree_3.12.0             RSpectra_0.16-2           irlba_2.3.5.1            
 [79] ggrastr_1.0.2             fastDummies_1.7.3         gridGraphics_0.5-1       
 [82] lazyeval_0.2.2            yaml_2.3.9                survival_3.7-0           
 [85] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [88] mapproj_1.2.11            progressr_0.14.0          tweenr_2.0.3             
 [91] later_1.3.2               ggridges_0.5.6            codetools_0.2-20         
 [94] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
 [97] Rtsne_0.17                shape_1.4.6.1             limma_3.60.4             
[100] estimability_1.5.1        filelock_1.0.3            pkgconfig_2.0.3          
[103] xml2_1.3.6                TMB_1.9.14                spatstat.univar_3.0-0    
[106] EnvStats_2.8.1            scatterplot3d_0.3-44      aplot_0.2.3              
[109] spatstat.sparse_3.1-0     ape_5.8                   xtable_1.8-4             
[112] highr_0.11                car_3.1-2                 plyr_1.8.9               
[115] httr_1.4.7                rbibutils_2.2.16          tools_4.4.1              
[118] globals_0.16.3            beeswarm_0.4.0            broom_1.0.6              
[121] nlme_3.1-165              HDO.db_0.99.1             dbplyr_2.5.0             
[124] lme4_1.1-35.5             digest_0.6.36             numDeriv_2016.8-1.1      
[127] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[130] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.1.5        
[133] glue_1.7.0                cachem_1.1.0              BiocFileCache_2.12.0     
[136] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[139] mvtnorm_1.2-5             presto_1.0.0              parallelly_1.37.1        
[142] statmod_1.5.0             RcppHNSW_0.6.0            ScaledMatrix_1.12.0      
[145] carData_3.0-5             minqa_1.2.7               pbapply_1.7-2            
[148] httr2_1.0.2               spam_2.10-0               gson_0.1.0               
[151] dqrng_0.4.1               utf8_1.2.4                graphlayouts_1.1.1       
[154] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.0      
[157] shiny_1.8.1.1             GenomeInfoDbData_1.2.12   glmmTMB_1.1.9            
[160] memoise_2.0.1             rmarkdown_2.27            scales_1.3.0             
[163] future_1.33.2             RANN_2.6.1                Cairo_1.6-2              
[166] spatstat.data_3.1-2       rstudioapi_0.16.0         cluster_2.1.6            
[169] spatstat.utils_3.0-5      hms_1.1.3                 fitdistrplus_1.2-1       
[172] munsell_0.5.1             cowplot_1.1.3             colorspace_2.1-0         
[175] rlang_1.1.4               xts_0.14.0                DelayedMatrixStats_1.26.0
[178] sparseMatrixStats_1.16.0  dotCall64_1.1-1           shinydashboard_0.7.2     
[181] ggforce_0.4.2             circlize_0.4.16           laeken_0.5.3             
[184] mgcv_1.9-1                xfun_0.46                 coda_0.19-4.1            
[187] e1071_1.7-14              TH.data_1.1-2             iterators_1.0.14         
[190] emmeans_1.10.3            abind_1.4-5               GOSemSim_2.30.0          
[193] treeio_1.28.0             bitops_1.0-7              Rdpack_2.6               
[196] promises_1.3.0            scatterpie_0.2.3          RSQLite_2.3.7            
[199] qvalue_2.36.0             sandwich_3.1-0            DelayedArray_0.30.1      
[202] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.1           
[205] prettyunits_1.2.0         boot_1.3-30               beachmat_2.20.0          
[208] listenv_0.9.1             Rcpp_1.0.13               edgeR_4.2.1              
[211] workflowr_1.7.1           BiocSingular_1.20.0       tensor_1.5               
[214] MASS_7.3-61               progress_1.2.3            BiocParallel_1.38.0      
[217] babelgene_22.9            spatstat.random_3.3-1     R6_2.5.1                 
[220] fastmap_1.2.0             multcomp_1.4-26           fastmatch_1.1-4          
[223] rstatix_0.7.2             TTR_0.24.4                vipor_0.4.7              
[226] ROCR_1.0-11               vcd_1.4-12                rsvd_1.0.5               
[229] nnet_7.3-19               gtable_0.3.5              KernSmooth_2.23-24       
[232] miniUI_0.1.1.1            deldir_2.0-4              ggthemes_5.1.0           
[235] htmltools_0.5.8.1         bit64_4.0.5               spatstat.explore_3.3-1   
[238] lifecycle_1.0.4           blme_1.0-5                nloptr_2.1.1             
[241] sass_0.4.9                vctrs_0.6.5               spatstat.geom_3.3-2      
[244] robustbase_0.99-3         ggfun_0.1.5               future.apply_1.11.2      
[247] bslib_0.7.0               pillar_1.9.0              pcaMethods_1.96.0        
[250] gplots_3.1.3.1            metapod_1.12.0            locfit_1.5-9.10          
[253] jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
